pkg/trace/api/endpoints.go

```diff
@@ -18,6 +22,9 @@ type endpoint struct {
 	// Hidden reports whether this endpoint should be hidden in the /info
 	// discovery endpoint.
 	Hidden bool
+
+	// ConfigEnabled reports whether this endpoing is enabled by a config
+	ConfigEnabled func(conf *config.AgentConfig) bool
```

Comment:
I think `IsEnabled` will be better
```suggestion
	// IsEnabled specifies a function which reports whether this endpoint should be enabled
	// based on the given config conf.
	IsEnabled func(conf *config.AgentConfig) bool
```